patches/0059-Fix-static-get-function-for-ssm.parameter-with-versi.patch

```diff
@@ -0,0 +1,23 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Florian Stadler <florian@pulumi.com>
+Date: Mon, 3 Jun 2024 21:23:04 +0200
+Subject: [PATCH] Fix static get function for ssm.parameter with versioned or
+ labled IDs
+
+When the SSM resources where moved to the AWS SDK v2 the lookup of
+SSM parameters with versions or labels got broken. This fixes that
+again.
+
+diff --git a/internal/service/ssm/parameter.go b/internal/service/ssm/parameter.go
+index 5a24c9641a..09d7cfea45 100644
+--- a/internal/service/ssm/parameter.go
++++ b/internal/service/ssm/parameter.go
+@@ -259,7 +259,7 @@ func resourceParameterRead(ctx context.Context, d *schema.ResourceData, meta int
+ 		return sdkdiag.AppendErrorf(diags, "invalid configuration, cannot set type = %s and insecure_value", param.Type)
+ 	}
+ 
+-	detail, err := findParameterMetadataByName(ctx, conn, d.Id())
```

Comment:
Before moving to SDK v2 it used the name. That's why the regression was introduced

Reply:
@flostadler that's very unfortunate. I'll add to our roadmap one more time the task of covering all patches with tests. It has to be done alas.